src/lightning/pytorch/utilities/data/__init__.py

```diff
@@ -0,0 +1,26 @@
+from lightning.pytorch.utilities.data.dataset import LightningDataset, S3LightningDataset
+
+# This is done for backwards compatibility
+from lightning.pytorch.utilities.data.extract_batch_size import (
+    _dataloader_init_kwargs_resolve_sampler,
+    _extract_batch_size,
+    _get_dataloader_init_args_and_kwargs,
+    _is_dataloader_shuffled,
+    _update_dataloader,
+    extract_batch_size,
+    has_len_all_ranks,
+    warning_cache,
+)
+
+__all__ = [
+    "extract_batch_size",
+    "_extract_batch_size",
```

Comment:
Why add protected to all and also why expose them?

Reply:
some of them are used in dataloader.py and other files. I just removed the private methods that aren't used in any other files.